### ISSUE TITLE
Boot order issue with hard drive disk image and CD image

### ISSUE THREAD

**Issue by rwv** [copy/v86#225]
I would like to boot freedos on hard disk image and use an iso file as a data drive. However, I encounter some strange behavior of the boot-order options. When both hard disk image and CD image are included. The hard drive boot entry will always be the last boot entry no matter the settings of boot order.
After more tests, it seems that if the CD image is include, the other two boot entry will be "disabled".

Order tables (with three entries both selected):

|Description|`boot_order`| First | Second | Third |
|---|---|-------|--------|-------|
|Hard Disk / CD / Floppy|`0x132`|CD|Floppy|hard disk| 
|Hard Disk / Floppy / CD|`0x312`|Floppy **(always fail)**|CD|hard disk| 
|Floppy / Hard Disk / CD|`0x321`|Floppy **(always fail)**|CD|hard disk| 
|Floppy / CD / Hard Disk|`0x231`|Floppy **(always fail)**|CD|hard disk| 
|CD / Hard Disk / Floppy|`0x123`|CD|Floppy|hard disk|
|CD / Floppy / CD|`0x213`|CD|Floppy|hard disk|

[disk image files](https://mega.nz/#!rq51iDAR!2Wz44VLSUFwEjAY8Ha6047jnOB8wo8jd57SD7yOHhvg) (upload to mega.nz because file size exceeds GitHub limits)

**Comment by copy** [comment]
Indeed, configurations with both CD and HDD are haven't been tested much.

I don't really have the time or energy to fix this, so contributions are very welcome.

**Comment by markostamcar** [comment]
Oh man, this bug really bums me out. It seems that using all 3 boot media types at the same time makes the boot fail completely.

I don't know much about coding an emulator but I did take a look at .js files and couldn't find anything like a bug in parsing boot media configuration etc... maybe it's a bug in the BIOS?

@rwv did you make any progress?
@copy what kind of bribe would tempt you to look into it - beer, 6 beers, 24 beers? :D

**Comment by rwv** [comment]
@markostamcar nope, i haven't made any progress.

**Comment by copy** [comment]
This is fixed now.
